File: java/src/main/java/com/pos/integration/service/PosService.java
package com.pos.integration.service;

import com.pos.integration.exception.PosException;
import com.pos.integration.model.auth.AuthRequest;
import com.pos.integration.model.auth.AuthResponse;
import com.pos.integration.model.redirect_url.RedirectUrlRequest;
import com.pos.integration.model.redirect_url.RedirectUrlResponse;
import com.pos.integration.model.refund.RefundRequest;
import com.pos.integration.model.refund.RefundResponse;

public interface PosService {

    RefundResponse refund(RefundRequest request) throws PosException;

    AuthResponse auth(AuthRequest request) throws PosException;

    RedirectUrlResponse fetchRedirectionUrl(RedirectUrlRequest request) throws PosException;
}

File: node/src/pos-integration/services/pos.service.ts
import { RefundRequest } from "../model/refund/refund-request";
import { AuthRequest } from "../model/auth/auth-request";
import { AuthResponse } from "../model/auth/auth-response";
import { RefundResponse } from "../model/refund/refund-response";
import { RedirectUrlResponse } from "../model/redirect-url/redirect-url-response";
import { RedirectUrlRequest } from "../model/redirect-url/redirect-url-request";

export interface PosService {
  refund(request: RefundRequest): Promise<RefundResponse>;

  auth(request: AuthRequest): Promise<AuthResponse>;

  fetchRedirectionUrl(request: RedirectUrlRequest): Promise<RedirectUrlResponse>;
}
